Kernel/API/MajorNumberAllocation.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2024, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Array.h>
+#include <AK/Platform.h>
+#include <AK/StringView.h>
+#include <AK/Types.h>
+#include <Kernel/API/DeviceFileTypes.h>
+
+namespace Kernel::MajorAllocation {
+
+enum class CharacterDeviceFamily : unsigned {
+    Generic = 1,
+    DeviceControl = 2,
+    Serial = 4,
+    Console = 5,
+    Mouse = 10,
+    FUSE = 11,
+    AllMice = 12,
+    GPURender = 28,
+    VirtualConsole = 35,
+    Keyboard = 85,
+    Audio = 116,
+    MasterPTY = 200,
+    SlavePTY = 201,
+    GPU = 226,
+    VirtIOConsole = 229,
+};
+
+static constexpr CharacterDeviceFamily s_character_device_numbers[] = {
+    CharacterDeviceFamily::Generic,
+    CharacterDeviceFamily::DeviceControl,
+    CharacterDeviceFamily::Serial,
+    CharacterDeviceFamily::Console,
+    CharacterDeviceFamily::Mouse,
+    CharacterDeviceFamily::FUSE,
+    CharacterDeviceFamily::AllMice,
+    CharacterDeviceFamily::GPURender,
+    CharacterDeviceFamily::VirtualConsole,
+    CharacterDeviceFamily::Keyboard,
+    CharacterDeviceFamily::Audio,
+    CharacterDeviceFamily::MasterPTY,
+    CharacterDeviceFamily::SlavePTY,
+    CharacterDeviceFamily::GPU,
+    CharacterDeviceFamily::VirtIOConsole,
+};
+
+constexpr bool assert_character_device_numbers_are_in_order()
+{
+    unsigned major = 0;
+    for (auto& allocation : s_character_device_numbers) {
+        if (to_underlying(allocation) < major)
+            return false;
+        major = to_underlying(allocation);
+    }
+    return true;
+}
+static_assert(assert_character_device_numbers_are_in_order());
+
+ALWAYS_INLINE MajorNumber character_device_family_to_major_number(CharacterDeviceFamily family)
+{
+    return static_cast<unsigned>(family);
+}
+
+ALWAYS_INLINE StringView character_device_family_to_string_view(CharacterDeviceFamily family)
```

Comment:
(nit: probably not hot; put outline in cpp file? suggestion for a future pr if you touch this file again. also for the other similar function further down.)